doc/content/integrations/pubsub/message-types/_index.md

```diff
@@ -0,0 +1,28 @@
+---
+title: "Message Types"
+description: ""
+---
+
+The Application Server can be configured to publish or subscribe to messages for any of the following events:
+- Uplink Message
+- Join Accept
+- Downlink Ack
+- Downlink Nack
+- Downlink Sent
+- Downlink Failed
+- Downlink Queued
+- Location Solved
+- Downlink Push
+- Downlink Replace
+
+ Enabling event messaging also allows you to manually configure a **Sub Topic** for that event. If no Sub Topic is specified, events will be published to the configured **Base Topic**.
+
+<!--more-->
+
+>Sub Topics should be specified for at least **Downlink Push** and **Downlink Replace** in order to schedule messages by publishing to these topics. If Downlink Push **AND** Downlink Replace are enabled, but have **no Sub Topic**, messages published to **Base Topic** may be scheduled as either Downlink Push **OR** Downlink Replace. This is an undesirable race condition.
```

Comment:
OK so what should the user know here really?

Reply:
hard to reword but i gave it a shot. if you wanna use both push and replace subtopics, you should specify subtopics for both. if you don't, and publish messages on the base topic, the application server will randomly schedule them as push or replace.